engine/src/test/java/io/zeebe/engine/job/JobTimeOutTest.java

```diff
@@ -15,58 +15,49 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package io.zeebe.broker.engine.job;
+package io.zeebe.engine.job;
 
 import static io.zeebe.protocol.intent.JobIntent.ACTIVATED;
 import static io.zeebe.protocol.intent.JobIntent.COMPLETED;
 import static io.zeebe.protocol.intent.JobIntent.FAILED;
 import static io.zeebe.protocol.intent.JobIntent.TIME_OUT;
+import static io.zeebe.test.util.record.RecordingExporter.jobRecords;
 import static org.assertj.core.api.Assertions.assertThat;
 
-import io.zeebe.broker.test.EmbeddedBrokerRule;
 import io.zeebe.engine.processor.workflow.job.JobTimeoutTrigger;
+import io.zeebe.engine.util.EngineRule;
 import io.zeebe.exporter.api.record.Record;
 import io.zeebe.exporter.api.record.RecordMetadata;
+import io.zeebe.exporter.api.record.value.JobBatchRecordValue;
 import io.zeebe.exporter.api.record.value.JobRecordValue;
-import io.zeebe.protocol.intent.Intent;
 import io.zeebe.protocol.intent.JobBatchIntent;
 import io.zeebe.protocol.intent.JobIntent;
-import io.zeebe.test.broker.protocol.commandapi.CommandApiRule;
-import io.zeebe.test.broker.protocol.commandapi.PartitionTestClient;
+import io.zeebe.test.util.record.RecordingExporter;
 import java.time.Duration;
 import java.util.List;
 import java.util.stream.Collectors;
-import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.RuleChain;
 
 public class JobTimeOutTest {
-  public EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
-  public CommandApiRule apiRule = new CommandApiRule(brokerRule::getAtomix);
+  private static final String PROCESS_ID = "process";
 
-  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(apiRule);
-  private PartitionTestClient client;
-
-  @Before
-  public void setup() {
-    client = apiRule.partitionClient();
-  }
+  @Rule public EngineRule engineRule = new EngineRule();
 
   @Test
   public void shouldNotTimeOutIfDeadlineNotExceeded() {
     // given
-    brokerRule.getClock().pinCurrentTime();
+    engineRule.getClock().pinCurrentTime();
     final Duration timeout = Duration.ofSeconds(60);
     final String jobType = "jobType";
 
-    createJob(jobType);
+    engineRule.createJob(jobType, PROCESS_ID);
 
-    apiRule.activateJobs(apiRule.getDefaultPartitionId(), jobType, timeout.toMillis()).await();
-    client.receiveFirstJobEvent(ACTIVATED);
+    engineRule.jobs().withType(jobType).withTimeout(timeout.toMillis()).activate();
+    jobRecords(ACTIVATED).getFirst();
 
     // when
-    brokerRule.getClock().addTime(timeout.minus(Duration.ofSeconds(1)));
+    engineRule.getClock().addTime(timeout.minus(Duration.ofSeconds(1)));
```

Comment:
There is a utility in the engine rule already for this. I had my reasons why I don't exposed the clock :see_no_evil: